### ISSUE TITLE
Mismatching of JSON data recieved

### ISSUE THREAD

**Issue by richp10** [nabbar/SwaggerValidator-PHP#20]
Amazing project.. I am however having one problem. 

I have swagger.json that passes the 'official' swagger validations, but I get a mismatching json error when loading with SwaggerValidator-php

```  \SwaggerValidator\Swagger::setSwaggerFile($file);
 $swagger = \SwaggerValidator\Swagger::load(new \SwaggerValidator\Common\Context());
```
Most of the file loads, then I get a error (TypeArrayItems.php is top of the list list) like this: 

``` [2017-04-17 20:00:04][EXCEPTION][{SwaggerValidator\DataType\TypeArrayItems::jsonUnSerialize#118] - Exception find : Mismatching type of JSON Data received --- {"contextMode":"DENY","contextType":"REQUEST","contextLocation":null,"contextScheme":null,"contextHost":null,"contextBasePath":null,"contextRequestPath":null,"contextRoutePath":null,"contextMethod":null,"contextDataPath":["#","paths","/groups","get","responses","200","schema","properties","data","items"],"contextDataCheck":[],"contextDataValue":[],"contextDataExists":null,"contextDataEmpty":null,"contextDecodeError":null,"contextDataType":null,"contextOther":[],"contextExternalRef":[],"contextIsCombined":null,"contextValidationCode":null,"mockedData":[]} ```

I think this is the portion json that is causing the problem - though as I say, it passes swagger validation:

``` "listgroups": {
            "required": [
                "status",
                "data"
            ],
            "properties": {
                "status": {
                    "type": "string",
                    "default": "Success"
                },
                "data": {
                    "type": "array",
                    "items": {
                        "properties": {
                            "uuid": {
                                "type": "string"
                            },
                            "groupname": {
                                "type": "string"
                            },
                            "grouphashname": {
                                "type": "string"
                            },
                            "orgtype": {
                                "type": "string"
                            },
                            "country": {
                                "type": "string"
                            },
                            "grouplogo": {
                                "type": "string"
                            },
                            "preferedlanguage": {
                                "type": "string"
                            },
                            "membercount": {
                                "type": "integer"
                            }
                        }
                    }
                }
            }
        },
```

Any thoughts? 

My use case is simply to validate swagger.json in a projects tests..

**Comment by nabbar** [comment]
I check it to control the array process
Maybe I forget to allow object without explicite "type=object"...

**Comment by richp10** [comment]
Thanks - if you can't find the issue with the extract provided, I can provide whole swagger.json..

**Comment by nabbar** [comment]
I confirm my suggest : 
```json
            "required": [
                "status",
                "data"
            ],
            "properties": {
                "status": {
                    "type": "string",
                    "default": "Success"
                },
                "data": {
                    "type": "array",
                    "items": {
                        "type": "object",
                        "properties": {
```

Normaly the spec tell that the type is mandatory, but I will add in next release the options to understand "properties" without type as implicit object.

**Comment by nabbar** [comment]
I have add an new test for this case. It will be check in every next release.
Some other question : what the php version you use ?

I thinking to stop the PHP 5.3, 5.4 and 5.5 compatibility

**Comment by richp10** [comment]
Many thanks for that. I use php 7.1 - I notice lots of libraries dropping 5.3 - but don't really have a view!

**Comment by richp10** [comment]
Thanks.. adding the type = "object" worked perfectly..

Though I then got to a different problem!

```
[2017-04-21 08:59:09][DECODE][{SwaggerValidator\Object\Paths::jsonUnSerialize#69] - Decoding Path "#/paths" As "SwaggerValidator\Object\Paths"
Call to undefined method SwaggerValidator\Exception::init()
 F:\WAPP\apps\rco-api2\vendor\njuhel\swagger-validator\src\Common\FactorySwagger.php:227
 F:\WAPP\apps\rco-api2\vendor\njuhel\swagger-validator\src\Common\FactorySwagger.php:265
 ```
There seems to be two issues here.. 
1) ThrowException itself raises the undefined method error.

2) The error itself is "Cannot identify the object builder for the given JSON Data" and I'm not sure what in the swagger.json is causing this.  

I put in a debug print_r and note that $originType and $OriginKey are blank when line 263 is called - also, context - contextDataPath is [0] => #

So it looks *to me* like a problem calling an empty or base path..

I thought it might help you to see my 'raw' swagger.json - please tell me when when you have grabbed this as I will then take down the gist. 

https://gist.github.com/richp10/bf3a940ef5d0cd7ebe03af6cf5562495

Thanks for your help on this - I hope tracking down these issues helps you hone the library!

oh, and this is the entirity of my code:

```
     $file = __DIR__ . '/../../app/swagger.json';
      include_once 'F:\WAPP\php5\SwaggerValidator.phar';
      \SwaggerValidator\Swagger::setSwaggerFile($file);
      $swagger = \SwaggerValidator\Swagger::load(new \SwaggerValidator\Common\Context());
```

**Comment by nabbar** [comment]
yes knwon : bug issue #18 
I will push a new patch this week end I think to fix it

**Comment by richp10** [comment]
Many thanks again..!

**Comment by richp10** [comment]
And *if* you have time, I still can't see why the exception was thrown.. my second issue..

**Comment by nabbar** [comment]
Fix in release 1.3.2 : more bug was fix following your swagger file

**Comment by nabbar** [comment]
If you find more errors, please start new issue :)
